components/dropdown/dropdown-content-mixin.js

```diff
@@ -683,7 +683,7 @@ export const DropdownContentMixin = superclass => class extends LocalizeCoreElem
 
 			const centerDelta = contentRect.width - targetRect.width;
 			const position = this._getPosition(spaceAround, centerDelta);
-			if (position) {
+			if (position !== null) {
```

Comment:
The implicit check is the root cause.  When the target wraps, the new position (left offset of content from the target's left edge) value is `0` or `-0`, and so this position assignment gets skipped, leaving the old rendered position unchanged in the DOM.